training/src/anemoi/training/utils/checkpoint.py

```diff
@@ -10,11 +10,17 @@
 from __future__ import annotations
 
 import logging
+import pickle  # noqa: S403  # pickle required for PyTorch checkpoint loading
 from pathlib import Path
 
 import torch
 import torch.nn as nn
 
+# Import pipeline exceptions for consistent error handling
+from anemoi.training.checkpoint.exceptions import CheckpointConfigError
```

Comment:
@bdvllrs also implemented some exception https://github.com/ecmwf/anemoi-core/blob/main/models/src/anemoi/models/migrations/migrator.py for the migration component. I some might attack different aspects but I wonder if we should just have a common script to store all of this and avoid duplications.